app/controllers/applications_controller.rb

```diff
@@ -14,6 +15,40 @@ def create
     end
   end
 
+  def edit
+    if cannot?(:edit, @application)
+      render :forbidden, status: 403
+      return
+    end
+  end
+
+  def update
+    if cannot?(:edit, @application)
```

Comment:
Should this be the `:update` permission?  I see other abilities using `:update` but I'm not sure if it's different for those or not.